app/javascript/application.js

```diff
@@ -7,3 +7,34 @@ window.jQuery = jquery
 window.$ = jquery
 
 Rails.start();
+
+document.addEventListener("DOMContentLoaded", function() {
+  
+  document.body.addEventListener("click", function(event) {
+    if (event.target.closest(".show-comments")) {
+      var icon = event.target.closest(".show-comments");
+      var postId = icon.getAttribute("data-comments");
+      var comments = document.getElementById(postId);
+
+      if (comments.style.display === "none" || comments.style.display === "") {
+        comments.style.display = "flex";
+      } else {
+        comments.style.display = "none";
+      }
+    }
+  });
+
+  document.body.addEventListener("click", function(event) {
+    if (event.target.closest(".show-comment-form")) {
+      var icon = event.target.closest(".show-comment-form");
+      var commentId = icon.getAttribute("data-comments-comment");
+      var comments = document.getElementById(commentId);
+
+      if (comments.style.display === "none" || comments.style.display === "") {
+        comments.style.display = "block";
+      } else {
+        comments.style.display = "none";
+      }
+    }
+  });
+});
```

Comment:
Maybe you can add conditions in order to reuse more logic and have a single function since they are very similar